quickwit/quickwit-common/src/lib.rs

```diff
@@ -174,6 +174,16 @@ where T: Debug
     }
 }
 
+#[inline]
+pub fn ceil_div(n: i64, q: i64) -> i64 {
+    assert!(q != 0i64);
+    // TODO trinity-1686a: i'm unsure about this assertion, i64::MIN makes more sense to me,
```

Comment:
good point.

the `div_ceil` in nightly goes

``
   pub const fn div_ceil(self, rhs: Self) -> Self {
            let d = self / rhs;
            let r = self % rhs;
            if (r > 0 && rhs > 0) || (r < 0 && rhs < 0) {
                d + 1
            } else {
                d
            }
        }
```

which is closer to you original value.

Reply:
So.... my bad. Also I am not sure what I suggested is exactly what you want for signed stuff. So forget my original comment and:
- do the implementation you see fit. performance is not an issue here.
- comment the behavior on negative int.
- maybe rename it ceil_div to align with rust